lib/pbench/server/database/models/datasets.py

```diff
@@ -24,13 +23,25 @@ class DatasetError(Exception):
     pass
 
 
+class DatasetBadName(DatasetError):
+    """
+    Specified filename does not follow Pbench tarball naming rules.
+    """
+
+    def __init__(self, name: Path):
+        self.name: str = str(name)
+
+    def __str__(self) -> str:
+        return f"File name {self.name!r} does not end in {Dataset.TARBALL_SUFFIX!r}"
+
+
 class DatasetSqlError(DatasetError):
     """
     SQLAlchemy errors reported through Dataset operations.
 
-    The exception will identify the controller and name of the target dataset,
-    along with the operation being attempted; the __cause__ will specify the
-    original SQLAlchemy exception.
+    The exception will identify the name of the target dataset, along with the
+    operation being attempted; the __cause__ will specify the original
+    SQLAlchemy exception.
```

Comment:
While I believe that what you say here is true, I don't see how the code in this class causes that to happen.  (That is, I think nearly all of that is effected by the code which instantiates this object, not by the methods of the object itself.)

_[I understand that all that this PR does is remove two words and re-wrap the rest, so if you're not inclined to address this, we can move along.]_

Reply:
True; we expect it to be raised in the context of a SQLAlchemy exception, but there's nothing this exception class can do about that. It's not without value, though it probably doesn't really belong here. Given the weight of approvals I'd just as soon "move along".